radix-engine/src/kernel/actor.rs

```diff
@@ -95,54 +109,33 @@ impl Actor {
                 ..
             }) => object_info.global,
             Actor::Function { .. } => true,
-            Actor::VirtualLazyLoad { .. } => false,
+            Actor::BlueprintHook { .. } => true,
             Actor::Root { .. } => false,
         }
     }
 
-    pub fn fn_identifier(&self) -> FnIdentifier {
+    pub fn fn_identifier(&self) -> Option<FnIdentifier> {
```

Comment:
It feels to me cleaner to mock the Root with a fake `FnIdentifier`/`BlueprintId`/`PackageAddress`/etc. rather than forcing the rest of the system to deal with the never used case of weird things happening at the root call frame. Thoughts?

Reply:
Unless there is substantial value in maintaining invariant, `FnIdentifier` is always available,  I'm leaning towards keeping the implementation simple and close to its true behaviour. 

This way we don't have to mock/reason about anything, and only need to follow compiler's guide and handle the extra case at consumers' side.